controllers/applicationvolumereplication_controller.go

```diff
@@ -0,0 +1,382 @@
+			logger.Error(err, "failed to find subscription list", "namespace", avr.Namespace)
+
+			return ctrl.Result{Requeue: true}, nil
+		}
+
+		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to list subscriptions")
+	}
+
+	placementDecisions := r.processSubscriptionList(subscriptionList)
+
+	if len(placementDecisions) == 0 {
+		logger.Error(nil, "no placement decisions were found", "namespace", avr.Namespace)
+
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if err := r.updateAVRStatus(ctx, avr, placementDecisions); err != nil {
+		logger.Error(err, "failed to update status")
+
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("completed manifestwork for subscriptions")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) processSubscriptionList(
+	subscriptionList *subv1.SubscriptionList) ramendrv1alpha1.SubscriptionPlacementDecisionMap {
+	placementDecisions := ramendrv1alpha1.SubscriptionPlacementDecisionMap{}
+
+	for _, subscription := range subscriptionList.Items {
+		// Create a ManifestWork that represents the VolumeReplicationGroup CR for the hub to
+		// deploy on the managed cluster. A manifest workload is defined as a set of Kubernetes
+		// resources. The ManifestWork must be created in the cluster namespace on the hub, so that
+		// agent on the corresponding managed cluster can access this resource and deploy on the
+		// managed cluster. We create one ManifestWork for each VRG CR.
+		homeCluster, peerCluster, err := r.createVRGManifestWork(subscription)
+		if err != nil {
+			r.Log.Info("failed to create or update ManifestWork for", "subscription", subscription.Name, "error", err)
+
+			continue
+		}
+
+		placementDecisions[subscription.Name] = &ramendrv1alpha1.SubscriptionPlacementDecision{
+			HomeCluster: homeCluster,
+			PeerCluster: peerCluster,
+		}
+
+		r.Log.Info("created VolumeReplicationGroup manifest for ", "subscription", subscription.Name,
+			"HomeCluster", homeCluster, "PeerCluster", peerCluster)
+	}
+
+	return placementDecisions
+}
+
+func (r *ApplicationVolumeReplicationReconciler) updateAVRStatus(
+	ctx context.Context,
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	placementDecisions ramendrv1alpha1.SubscriptionPlacementDecisionMap) error {
+	r.Log.Info("updating AVR status")
+
+	avr.Status = ramendrv1alpha1.ApplicationVolumeReplicationStatus{
+		Decisions: placementDecisions,
+	}
+	if err := r.Client.Status().Update(ctx, avr); err != nil {
+		return errorswrapper.Wrap(err, "failed to update AVR status")
+	}
+
+	return nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) createVRGManifestWork(
+	subscription subv1.Subscription) (string, string, error) {
+	const empty string = ""
+
+	// Select cluster decisions from each Subscriptions of an app.
+	homeCluster, peerCluster, err := r.selectPlacementDecision(&subscription)
+	if err != nil {
+		return empty, empty, fmt.Errorf("failed to get placement targets from subscription %s with error: %w",
```

Comment:
The wrapping is done by fmt.Errorf.  We could use that.  I will look at the error wrap more closely.  I was trying to satisfy the linter where each time it sees that an error is returned from an external interface, it complains that you should wrap it so that you can add more context to it.  If however I change line 166 to something like:
```
return r.Client.Status().Update(ctx, avr)
```
then the linter will not complain